### ISSUE TITLE
Performance issue, nvim + osx

### ISSUE THREAD

**Issue by reynn** [zellij-org/zellij#296]
Really enjoying the project so far awesome work!

I am seeing some performance issues when running NVIM in Zellij, when scrolling rapidly through a file all of Zellij starts lagging. In another tab this runs pretty smoothly. I've been able to replicate the issue in [Alacritty](https://github.com/alacritty/alacritty), iTerm2 and the builtin OSX terminal.

The first video here is how scrolling looks on the machine without going through Zellig, second is within.

Let me know how I can help troubleshoot or with doing rust debugging, if you need it.

https://user-images.githubusercontent.com/224374/115497136-ec797200-a21f-11eb-9bc8-605cc79513df.mov

https://user-images.githubusercontent.com/224374/115497143-f0a58f80-a21f-11eb-8c8b-697fcb10495c.mov

**Comment by keeslinp** [comment]
I saw similar issues. Might be worthwhile to see if shrinking the alacritty window makes it faster. It sounds like the whole window gets redrawn every frame so the larger the window the more has to be "touched" in each render.

**Comment by TheAlgorythm** [comment]
I have the same problems on macOS on [bat](https://github.com/sharkdp/bat)/less and even tipping in the command prompt with [autosuggestions](https://github.com/zsh-users/zsh-autosuggestions) at fullscreen with 2560 × 1600 pixels.
If I decrease the size, the effect is significantly less (pun intended) noticeable.

**Comment by imsnif** [comment]
@reynn - while Zellij has some performance problems (as @keeslinp mentioned) this looks especially bad and I'm afraid you hit some sort of hidden edge-case that I'm not (and want to at least be) aware of.

Would you be willing to help me out debug this? I can't reproduce this with neovim on my box, so it might be something particular to your neovim setup. If you follow the instructions here and attach the log files (especially the pane logs, eg. zellij-3.log), I'd probably be able to use them to re-create the issue on my machine: https://github.com/zellij-org/zellij/blob/main/.github/ISSUE_TEMPLATE/bug_report.md

With the `--debug` flag, just scroll down once and then quit zellij. That should be enough for me to see this as well.
Much appreciated!

**Comment by raphCode** [comment]
Hey, is still happening with current zellij versions?
@TheAlgorythm @keeslinp @reynn

**Comment by TheAlgorythm** [comment]
It is definitely much better. It's maybe a little more stuttering than without it, but it is nearly unnoticeable.

**Comment by keeslinp** [comment]
I would agree that it is a lot better (I'm also on a faster computer now), but it is definitely still noticeable compared to just using kitty tabs or tmux.

**Comment by imsnif** [comment]
I have a tiny suspicion, I wonder if you'll be willing to help me check it?
Does this happen if the background of neovim is the same color as the background of your terminal?

Otherwise, if one of you would be willing to attach debug logs using the bug report template: https://github.com/zellij-org/zellij/pull/1358 it would be very helpful (please also be sure to include the `tput lines` and `tput cols` output inside the pane before opening neovim). That way I can reproduce this on my own machine and find out exactly what the issue is.

**Comment by imsnif** [comment]
Anyone experiencing this issue and willing to help debug? For the life of my I can't reproduce it no matter what I try :) Maybe @tlinford ?

**Comment by raphCode** [comment]
I looked into profiling and I believe `perf` could probably yield some insights where time is spent:

Make a debug build, run it and set up the reproducer with the performance issues, then run:
`perf record --stat -g --pid=$(pgrep zellij | paste -sd ,)`
Please make sure there is only one zellij session and a single client, otherwise specify the pids of the relevant processes (server and client) manually: `--pid=1,2,1337`.
While this is recording, trigger the performance problem inside zellij.
Stop recording with Ctrl-C.

Finally, package up the recorded data with
`perf archive`
and upload the file here (should only have a couple of MB)

**Comment by quolpr** [comment]
I decided to compare performance of fast scroll with tmux, and zellij is slugish here. Watch the video:

https://github.com/zellij-org/zellij/assets/7958527/c4cacd2f-e01d-4880-8d97-fc1958bcaebc

Here how it works with tmux:

https://github.com/zellij-org/zellij/assets/7958527/aa7a5c65-1672-4bd0-b68d-91233bdff74e

It struggles too, but still, it's not as sluggish as with zellij.

I am using alacritty + latest zellij with default config on macos 14.5 m1 max

**Comment by rednaks** [comment]
any updates on this ? it gets worse :)
